### PR TITLE
Set C++ standard to c++14

### LINKED ISSUE #336: Set C++ standard to c++14 for all packages
```cmake
if(NOT CMAKE_CXX_STANDARD)
  set(CMAKE_CXX_STANDARD 14)
endif()
```

### PR DESCRIPTION
**Pull request by hsd-dev**
Closes #336

### CODE REVIEW

File: can_msgs/CMakeLists.txt [hunk 1]
```diff
@@ -1,6 +1,10 @@
 cmake_minimum_required(VERSION 2.8.3)
 project(can_msgs)
 
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
```
Comment (gavanderhoorn) [inline can_msgs/CMakeLists.txt:7]: Up to @ipa-mdl, but would you perhaps want to add `set(CMAKE_CXX_STANDARD_REQUIRED ON)` here as well?

Comment (mathias-luedtke) [reply can_msgs/CMakeLists.txt:7]: @gavanderhoorn: Thanks for pointing this out.

**Review by mathias-luedtke — CHANGES_REQUESTED** [review]
Please add `set(CMAKE_CXX_STANDARD_REQUIRED ON)` to all packages

**Comment by hsd-dev** [comment]
@ipa-mdl please take a look.

File: can_msgs/CMakeLists.txt [hunk 2]
```diff
@@ -1,6 +1,11 @@
 cmake_minimum_required(VERSION 2.8.3)
 project(can_msgs)
 
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
```
Comment (mathias-luedtke) [inline can_msgs/CMakeLists.txt]: I would put that inside the clause..

**Review by mathias-luedtke — CHANGES_REQUESTED** [review]

File: can_msgs/CMakeLists.txt (hunk 2 above)
Comment (hsd-dev) [reply can_msgs/CMakeLists.txt]: Done

**Review by mathias-luedtke — APPROVED** [review]
:+1:

**Pull request merged**